src/util/macro_util.rs

```diff
@@ -649,7 +749,7 @@ where
 {
     match try_cast_or_pme::<Src, Dst, _, BecauseExclusive>(Ptr::from_mut(src)) {
         Ok(ptr) => {
-            static_assert!(Src, Dst => mem::align_of::<Dst>() <= mem::align_of::<Src>());
```

Comment:
This changes the semantics of the function. It's important that the assertion as at post-monomorphization time.